### PR TITLE
Juniper: add filter term/action

### PR DESCRIPTION
**Pull request by XioNoX**
Fixes issues #246.

### CODE REVIEW

**Comment by XioNoX** [comment]
Not sure how to fix that failing test. Guidance welcome :)

**Comment by XioNoX** [comment]
@abhindes pinging you as you're assigned to the task. Anything more I can do here? Thanks

**Comment by abhindes** [comment]
You could rename it to "filter-term" since that is what it is. "nested-filter" also works.

**Comment by XioNoX** [comment]
@abhindes indeed, thanks! updated.

**Comment by abhindes** [comment]
Hi, could you also update the .pol file with an example of a term using "filter-term".

I just wanted to check the generated diff of the .pol file for Juniper, so that we can see the changes from this PR. 

Thanks!

**Comment by XioNoX** [comment]
@abhindes is it ready to be merged?

**Comment by XioNoX** [comment]
@abhindes is there anything more I need to do? Thanks

**Comment by XioNoX** [comment]
Hi @abhindes is there anything more I can do? Thanks

**Comment by XioNoX** [comment]
@abhindes @kevinsteves @rdsharma could someone from Capirca have a look at my 3 PRs? Thanks

File: tests/lib/srxlo_test.py [hunk 1]
```diff
@@ -81,6 +81,7 @@
     'dscp_set',
     'ether_type',
     'expiration',
+    'filter_term',
```
Comment (abhindes) [inline tests/lib/srxlo_test.py:84]: Does the SRX support hierarchical filters? 

I can't quickly find documentation either way.

Comment (abhindes) [reply tests/lib/srxlo_test.py:84]: This is called Nested filters and it doesn't seem like SRX supports this. 

https://apps.juniper.net/feature-explorer/feature-info.html?fKey=10519&fn=Nested%20Filters

Comment (abhindes) [reply tests/lib/srxlo_test.py:84]: The changes otherwise look great! Thanks!

Comment (XioNoX) [reply tests/lib/srxlo_test.py:84]: If my understanding is correct, the `srxlo` generator is for firewall filters on SRX devices, which is different from the security policies of the `srx` target.
I tested the following on a SRX220 and it accepted the commit:
```
[edit]
+  firewall {
+      family inet {
+          filter test {
+              term foo {
+                  filter bar;
+              }
+          }
+          filter bar {
+              term accept {
+                  then accept;
+              }
+          }
+      }
+  }
```
If those assumptions are correct, I think it's fully supported. And different from nested filters, which only seem to be present on high end PTX (with no mention of MXs where I tested them as well).

Comment (abhindes) [reply tests/lib/srxlo_test.py:84]: Yup, srxlo seems to be the loopback on SRX devices. Interesting, thanks for testing this out!

**Review by abhindes — APPROVED** [review]

**Comment by abhindes** [comment]
Hi,

I'm having some trouble trying to merge this in (there's some bug being encountered and I've tried different workarounds; I've raised a bug for this but don't know when it'll get resolved). 

Could you please send a clone of this PR as a new PR, which I can quickly approve and try merging it afresh. Thanks, and I apologize for the inconvenience.

**Comment by XioNoX** [comment]
> Could you please send a clone of this PR as a new PR, which I can quickly approve and try merging it afresh.

No pb at all, please see https://github.com/google/capirca/pull/293.

Would it be possible to do a new Capirca release once all are merged? Thanks

**Comment by abhindes** [comment]
Merged, and there's a new release! Thanks, closing this out.

**Pull request closed without merging**